src/licensedcode/plugin_license.py

```diff
@@ -116,3 +119,57 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
+    """
+    Find instances for any licenses in referenced filenames
+    """
+    if not resource.is_file:
+        return
+
+    licenses = resource.licenses
+    license_expressions = resource.license_expressions
+    if not licenses:
+        return 
+
+    reference_licenses=[]
+    reference_license_expressions=[]
+    unique_files = []
+
+    for license in licenses:
+        ref_files=license['matched_rule']['referenced_filenames']
+        if len(ref_files) != 0:
+            for i in range(len(ref_files)):
+                if ref_files[i].startswith('usr/share/common-licenses') or ref_files[i] in unique_files:
```

Comment:
What's unique files?
Could you just instead ensure that all rules referenced_filenames are unique at construction and validation time?

Reply:
umm didn't got your point ;( . Suppose there are two licenses in licenses attribute and both of them reference to same file we need to report it only once right?

Reply:
I have made it now separate process instead of having it nested